samza-api/src/main/java/org/apache/samza/table/descriptors/BaseTableDescriptor.java

```diff
@@ -78,33 +68,33 @@ public String getTableId() {
     return tableId;
   }
 
-  /**
-   * Get the serde assigned to this {@link TableDescriptor}
-   *
-   * @return {@link KVSerde} used by this table
-   */
-  public KVSerde<K, V> getSerde() {
-    return serde;
+  @Override
+  public Map<String, String> toConfig(Config jobConfig) {
+
+    validate();
+
+    Map<String, String> tableConfig = new HashMap<>(config);
+    tableConfig.put(
+        String.format(JavaTableConfig.TABLE_PROVIDER_FACTORY, tableId),
+        getProviderFactoryClassName());
+    generateConfig(jobConfig, tableConfig);
+    return Collections.unmodifiableMap(tableConfig);
   }
 
-  /**
-   * Generate config for {@link TableSpec}; this method is used internally.
-   * @param tableSpecConfig configuration for the {@link TableSpec}
-   */
-  protected void generateTableSpecConfig(Map<String, String> tableSpecConfig) {
-    tableSpecConfig.putAll(config);
+  public String getProviderFactoryClassName() {
+    throw new SamzaException("Not implemented");
```

Comment:
Is the idea that extending classes should provide this? If so, maybe make this an abstract method instead?